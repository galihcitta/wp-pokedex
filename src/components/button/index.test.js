import React from "react";
import Button from "./index";

describe("Button Component", () => {
    it("renders and create snapshot <Button />", () => {
        expect(
            toJson(shallow(<Button block type="button"></Button>)),
        ).toMatchSnapshot();
    });

    it("check children have text <Button />", () => {
        expect(
            shallow(
                <Button block type="button">
                    Title
                </Button>,
            ).text(),
        ).toEqual("Title");
    });
    it("check children have text <Button />", () => {
        const tree = renderer
            .create(
                <Button block type="button">
                    Title
                </Button>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree).toHaveStyleRule("display", "block");
    });
    it("check children have text <Button />", () => {
        const tree = renderer
            .create(<Button type="button">Title</Button>)
            .toJSON();
        expect(tree).toHaveStyleRule("display", "inline-block");
    });
});
