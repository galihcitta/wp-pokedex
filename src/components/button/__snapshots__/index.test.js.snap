// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component check children have text <Button /> 1`] = `
.c0 {
  background: #fff;
  border: 1px solid transparent;
  border-radius: 8px;
  color: #fff;
  cursor: pointer;
  display: block;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  outline: none;
  padding: 0.375rem 0.75rem;
  position: relative;
  text-align: center;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  white-space: nowrap;
  width: auto;
}

<button
  className="c0"
  type="button"
>
  Title
</button>
`;

exports[`Button Component renders and create snapshot <Button /> 1`] = `
.c0 {
  background: #fff;
  border: 1px solid transparent;
  border-radius: 8px;
  color: #fff;
  cursor: pointer;
  display: block;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
  outline: none;
  padding: 0.375rem 0.75rem;
  position: relative;
  text-align: center;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out, border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  white-space: nowrap;
  width: auto;
}

<button
  className="c0"
  type="button"
/>
`;
